### DISCUSSION TITLE
Random 502 error for graphql queries

### DISCUSSION THREAD

**Discussion by agsmartwolf** [swellstores/community#165] [Help] [answered]
Hi mates,

I'm using a horizon theme - cloned and forked it.
During the build process - there is often a **502** error for the query `getStoreSettings` for the product page.
It happens randomly. Sometimes there are errors - sometimes not.

I think it's an issue with your infrastructure. It would be great if you could investigate this problem.
I have to restart my build pipelines 2-7 times to get them to succeed.

**Comment by agsmartwolf** [comment]
I have 502 error VERY often. Could you please explain what problem is in?

Here is a screen from vercel pipeline builder:

<img width="1400" alt="Screenshot 2023-03-28 at 17 20 27" src="https://user-images.githubusercontent.com/109105178/228250018-e5c2138e-d2ef-438b-99c2-10b6b8a00d79.png">

**Reply by ericingram** [reply, marked as answer]
@agsmartwolf This is probably caused by our infrastructure which is split across several cloud providers. We're currently working on merging to a single provider but it could take another ~30 days or so to get there. At that point we'll have better observability and would expect this kind of issue to become rare or non-existant

I'm sure it's not the answer you're hoping for but it might help to see the light at the end of the tunnel

**Reply by agsmartwolf** [reply]
Thanks for your reply. I hope everything works out for you - for all of us)
